GSA-TTS/FAC
# Prevalidation Summaries

# Prevalidation Summaries

This draft PR covers a lot of ground. It may be worth breaking into smaller pieces. 

Implements several line items in epic: https://github.com/GSA-TTS/FAC/issues/2655
Based on PR: https://github.com/GSA-TTS/FAC/pull/2934

## Changes:
1. IntakeToDissemination has minor changes to allow us to run an almost complete audit through (pre-certification), to prep the data for a bulk download.
2. Summary report generation. There are two versions. 
    * One works on the dissemination tables. It can handle one or many audits, and will later be used for both dissemination summary reports and results table bulk summary reports. 
    * One works on pre-certification IntakeToDissemination data. This works on only one audit, and is used for a pre-cert report download.
4. Several new URLs & views, for downloading files related to an in-progress submission. 
    * PDF report download
    * XLSX download (any kind)
    * XLSX download for summary reports. This will generate the report, put it in s3, and serve it to the user. There is one for both pre-cert and dissemination, and they use the data generators. 
4. Certification pages
    * Include links to download the summary and the PDF.
5. Dissemination summaries have a new link to the XLSX summary report for that record.

## How to test:
1. Switch to this branch and run normally. Requires a docker re-up for me.
6. Start a submission, confirm that the checklist doesn't show a review link.
7. Step through the submission and lock for certification. Confirm that:
    * The review link exists
    * The review page (summary) looks right
    * Normal files can be downloaded
    * The XLSX download works properly (wraps all submission data into one file)
8. Navigate to search. Find a submitted audit, and view the summary. Verify that the XLSX download works properly. 

## PR checklist: submitters


## PR checklist: reviewers


The larger the PR, the stricter we should be about these points.
